src/coreclr/vm/prestub.cpp

```diff
@@ -1059,7 +1045,444 @@ PCODE MethodDesc::JitCompileCodeLocked(PrepareCodeConfig* pConfig, JitListLockEn
+        {
+            MethodDesc* curr = iter.GetDeclMethodDesc();
+
+            // Check the target and current method match static/instance state.
+            if (cxt.IsTargetStatic != (!!curr->IsStatic()))
+                continue;
+
+            // Check for matching name
+            if (strcmp(methodName, curr->GetNameThrowing()) != 0)
+                continue;
+
+            // Reset reading the declaration signature
+            cxt.DeclarationSig.Reset();
+
+            MetaSig currSig(curr);
+
+            // Validate calling convention.
+            if (cxt.DeclarationSig.GetCallingConvention() != currSig.GetCallingConvention())
+                continue;
+
+            // Validate the return type and prepare for validating
+            // the current signature's argument list.
+            UINT argCount = cxt.DeclarationSig.NumFixedArgs();
+            if (cxt.Kind == UnsafeAccessorKind::Constructor)
+            {
+                // Constructors must return void.
+                if (currSig.GetReturnType() != ELEMENT_TYPE_VOID)
+                    continue;
+            }
+            else
+            {
+                declCorType = cxt.DeclarationSig.GetReturnTypeNormalized(&declType);
+                maybeCorType = currSig.GetReturnTypeNormalized(&maybeType);
+                if (declCorType != maybeCorType)
+                    continue;
+                if (declType != maybeType)
+                    continue;
+
+                // Non-constructor accessors skip the first argument
+                // when validating the target argument list.
+                cxt.DeclarationSig.SkipArg();
+                argCount--;
+            }
+
+            // Validate argument count matches.
+            if (argCount != currSig.NumFixedArgs())
+                continue;
+
+            // Validate arguments match.
+            for (; argCount > 0; --argCount)
+            {
+                declCorType = cxt.DeclarationSig.PeekArgNormalized(&declType);
+                maybeCorType = currSig.PeekArgNormalized(&maybeType);
+                if (declCorType != maybeCorType)
+                    break;
+                if (declType != maybeType)
+                    break;
+                cxt.DeclarationSig.NextArg();
+                currSig.NextArg();
+            }
+
+            // If we validated all arguments, we have a match.
+            if (argCount != 0)
+                continue;
+
+            cxt.TargetMethod = curr;
+            return true;
+        }
+        return false;
+    }
+
+    bool TrySetTargetMethodCtor(GenerationContext& cxt)
+    {
+        STANDARD_VM_CONTRACT;
+        _ASSERTE(!cxt.TargetType.IsNull());
+        _ASSERTE(cxt.Kind == UnsafeAccessorKind::Constructor);
+
+        TypeHandle targetType = cxt.TargetType;
+        _ASSERTE(!targetType.IsTypeDesc());
+
+        PTR_MethodTable pMT = targetType.GetMethodTable();
+
+        // Special case the default constructor case.
```

Comment:
Is this just a perf optimization or is this needed for correctness?

Reply:
It was an attempt at me being clever, but over the past few iterations it has been more of a pain than I thought. @MichalStrehovsky is also asking about it. Any concerns with me just removing it?

Reply:
If it is not needed for correctness, I think it would be better to remove it.